### ISSUE TITLE
Crash with raw types and lambdas

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#8169]
```java
import java.util.List;

/**
 * A lambda whose target type is a raw generic functional interface, returning the result of a
 * generic method whose invocation is an unchecked conversion: a raw {@code List} is passed where
 * {@code List<F>} is expected, so the result is the erased {@code List}.
 *
 * <p>Inference then creates a constraint from that erased {@code List} to the target type, the type
 * variable {@code OutputT}. The bound is false, so inference throws FalseBoundException and the
 * Checker Framework crashes in DefaultTypeArgumentInference.inferTypeArgs.
 *
 * <p>Run with -AconvertTypeArgInferenceCrashToWarning=false, which is what the Beam build sets;
 * otherwise the crash is reported as a warning instead.
 *
 * <p>Reduced from org.apache.beam.sdk.io.aws2.schemas.AwsTypes.transformList, which is
 * {@code return list -> Lists.transform((List) list, fn::apply);}. Neither the second argument nor
 * the method reference is needed to trigger the crash.
 */
public class AwsTypesCrash {

  interface SerializableFunction<InputT, OutputT> {
    OutputT apply(InputT input);
  }

  static <F> List<F> transform(List<F> from) {
    throw new UnsupportedOperationException();
  }

  @SuppressWarnings({"rawtypes", "unchecked"})
  private static SerializableFunction transformList() {
    return list -> transform((List) list);
  }
}

```
